packages/components/src/surface/README.md

```diff
@@ -24,48 +27,58 @@ function Example() {
 
 ## Props
 
-##### backgroundSize
+### `backgroundSize`: number
 
-**Type**: `number`
+- Required: No
+- Default: `12`
 
 Determines the grid size for "dotted" and "grid" variants.
 
-##### border
+### `border`: `boolean`
```

Comment:
I'm not really sure about the current APIs around `border`.

Currently, there are 5 different props:

- `border` shows the border on all four sides
- `border{Top,Left,Right,Bottom}` shoes a border on each side individually

It feels a bit weird to me that there is partial overlap between them — i.e. that if `border` is `true`, all other props become meaningless

Reply:
Agreed, this doesn't make much sense to me, especially if we consider how borders behave in CSS. I guess it makes sense that the `border*` take precedence over `border`, but that might also require a change in the default values of those props.

Reply:
Regardless of what we decide here, it makes sense to document how all those border props behave, and which one takes precedence over the other(s).

Reply:
Probably the easiest solution that I see is to remove the `border` prop, and leave the remaining individual props as they are. That would avoid overlaps and only make setting a border around the whole surface a bit more verbose.

WDYT?

Reply:
I'd say I can see some value in having them separate, but I'd likely have the `border*` default to the current value of `border` of not specified. I think that would be easy, too, and would solve the problem, while still resembling how border specificity works in CSS. What do you think about that?